.github/workflows/check_code_style.yml

```
@@ -0,0 +1,23 @@
+name: Check Code Style
+
+on:
+  push:
+    branches:
+      - "**"
+  pull_request:
+    branches:
+      - "main" 
+
+jobs:
+  check_code_style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Initialize and update submodule
+        run: git submodule update --init
```

Should we also be recursive?
```suggestion
        run: git submodule update --init --recursive
```

---

Yep, I think this is a great idea!